src/aap_eda/services/ruleset/activate_rulesets.py

```diff
@@ -126,6 +126,14 @@ def activate(self, activation: models.Activation) -> None:
 
             activation_db_logger = ActivationDbLogger(instance.id)
 
+            if not activation.user.get_awx_tokens().exists():
+                instance.status = ActivationStatus.FAILED
```

Comment:
@Alex-Izquierdo is this going to trigger a restart because of Restart Policy is failed. We would have to prevent a restart.

Reply:
@mkanoor From my testing, it didn't trigger a restart with `on-failure` restart policy.

Reply:
@mkanoor I thought on that as well, but I finally thought that we keep the restart logic simpler if we treat it in the same way, because in the end is another failed activation. Keep it simple principle. Adding a token is quite quick and it is something that in the most of the cases will happen because the user has forgotten it. 

To me the user story is:
- oh, there is no token, well, the activation will be restarted, I will add it to make it working the next attempt. 

It is basically the same case as if you use a rulebook with a job_template that does not exist, the activation continues its flow and at any time you can add that job_template to AWX so that everything works without you having to do anything else. 

Also keep in mind that if the user does not notice the problem at the time it occurs, the activation will still reach the limit of attempts and you will still have to manually restart it after adding the token. 

So, to end up in the same place, if we can give the user the opportunity to not have to restart the activation and just add the token, more convenient for the user and better UX. To me is not different than another failed activation scenario. 

what do you think @hsong-rh @bzwei

Reply:
@Dostonbek1 that is true, right now it is only restarted manually, but I think this is because we don't have yet the restart logic 100% fully working, @hsong-rh is working on that.

Reply:
@Alex-Izquierdo We would be relying too much on the restart logic to handle user errors and the timing issues, when the monitor picks up the task. It's much simpler to say this is a well known error, the pre-reqs for running an activation have not been satisfied so we should just mark it as failed and not restart it. The same thing happens when a DE is missing we can't say mark it as failed and hope that the user can get in between the 5 restarts and sneak in a DE. They can just restart it once they have fixed the problem. This also creates a race conditions between when the user sets the value and when the monitor picks up the task leading to more confusion.

Reply:
@mkanoor The DE is not the same case, because the DE is bind to the activation. Anyway, that requires to know in advance the restart workflow, so I see two options: or we implement this as it is and we change it with the modifications in the restart workflow or I wait to have it implemented to update the PR. I will talk with @hsong-rh to be aligned on this.